src/core/dev/ib_ctx_handler.h

```diff
@@ -34,106 +34,49 @@
 #ifndef IB_CTX_HANDLER_H
 #define IB_CTX_HANDLER_H
 
-#include <infiniband/verbs.h>
-#include <unordered_map>
+#include "dpcp/ib_ctx_handler_ibv.h"
+#include "doca/ib_ctx_handler_doca.h"
 
-#include "event/event_handler_ibverbs.h"
-#include "dev/time_converter.h"
-#include "ib/base/verbs_extra.h"
-#include "utils/lock_wrapper.h"
-#include <mellanox/dpcp.h>
-#include <doca_dev.h>
-#include <doca_flow.h>
-#include <doca_pe.h>
-
-typedef std::unordered_map<uint32_t, struct ibv_mr *> mr_map_lkey_t;
-
-struct pacing_caps_t {
-    uint32_t rate_limit_min;
-    uint32_t rate_limit_max;
-    bool burst;
-
-    pacing_caps_t()
-        : rate_limit_min(0)
-        , rate_limit_max(0)
-        , burst(false) {};
-};
-
-// client to event manager 'command' invoker (??)
-//
-class ib_ctx_handler : public event_handler_ibverbs {
+class ib_ctx_handler {
```

Comment:
that was hard to follow - so ib_ctx_handler is not a base class - but a conditional holder of doca or dpcp?

Consider choosing an easier abstraction - like inheritance - and compile out the child classes (and files) based on compilation flags in the makefile

Reply:
Discussed